### PR TITLE
Improve training config

### PR DESCRIPTION
**Pull request by pszemraj**
This PR aims to bring improvements to the training args used for pretraining MPNet from several angles:

1. improved default values for the training args[^1] and updates to some to more closely follow hyperparams in MPNet paper
2. clearer, more succinct descriptions of what the core args are/do and how to use them
3. addition of new A) options for some existing training args[^2] and B) exposing/integrating some hardcoded parameters[^3] to new CLI args to be adjustable by the user 

[^1]: i.e. like grad clip which has become standard during pretrain since the original repo came out
[^2]: added support for new activation fns "silu" and "relu2"
[^3]: the relaative attention hyperparams `relative_attention_num_buckets` and `max_distance` are hardcoded to values for 512 ctx, dhould be set-able by user w/ reasonable defaults

### CODE REVIEW

**Comment by pszemraj** [comment]
while the main desc covers the changes conceptually, from a more practical perspective of work items I'm thinking about it as:

1. update relevant training arg default values, add some new options to a few, update their descriptions.
2. the context length fiasco: add support for overriding default relative attn hyperparams `relative_attention_num_buckets` and `max_distance` via CLI, need to integrate through to modeling code.
    - Additionally, the default values of relative attn params will _automatically scale relative to the context length trained at_
3. support moving the tokenizer repo id to a training arg and using/saving with any tokenizer (provided it has the right special tokens, behavior, etc)

item 1 is mostly done, just needs some quick tests. rest are WIP

**Review by amazingvince — DISMISSED** [review]

**Comment by pszemraj** [comment]
for item 2: setting/computing `relative_attention_num_buckets` and `max_distance`:

- auto-setting works, setting/overriding via CLI works
- training works (both at 512 and 1024)
- converting saved checkpoint works

**Comment by pszemraj** [comment]
note that because of [not token.startswith("##")](https://github.com/pszemraj/annotated-mpnet/blob/660b6e2aaba1550de9beb16dc97a6bdec441759f/annotated_mpnet/data/mpnet_data.py#L351) custom tokenizer support is 'initial'/has to be a wordpiece tokenizer to do whole word masking. Currently in the train loop I have it raise an error if it's not a wordpiece tokenizer to be on the safe side.

To use a non-wordpiece tokenizer, like [modernbert's](https://huggingface.co/answerdotai/ModernBERT-base), would need to 1. comment out/remove [is_valid and details["whole_word_mask"]](https://github.com/pszemraj/annotated-mpnet/blob/660b6e2aaba1550de9beb16dc97a6bdec441759f/cli_tools/pretrain_mpnet.py#L149) and 2. [disable whole word masking in the collator when it's loaded](https://github.com/pszemraj/annotated-mpnet/blob/660b6e2aaba1550de9beb16dc97a6bdec441759f/cli_tools/pretrain_mpnet.py#L182). 

we can look at supporting whole word masking for more tokenizer types in a future PR

**Review by amazingvince — APPROVED** [review]
